go.py

```diff
@@ -236,31 +234,31 @@ def add_stone(self, color, c):
 
         return captured_stones
 
-    def _create_group(self, color, c, liberties):
+    def _merge_from_played(self, color, played, libs, other_group_ids):
+        stones = set()
+        liberties = set(libs)
+        for other_group in other_group_ids:
+            other = self.groups.pop(other_group)
+            stones.update(other.stones)
+            liberties.update(other.liberties)
+
+        assert stones.isdisjoint(liberties)
         self.max_group_id += 1
-        new_group = Group(self.max_group_id, frozenset([c]), liberties, color)
-        self.groups[new_group.id] = new_group
-        self.group_index[c] = new_group.id
-        self.liberty_cache[c] = len(liberties)
-        return new_group
-
-    def _merge_groups(self, group1_id, group2_id):
-        group1 = self.groups[group1_id]
-        group2 = self.groups[group2_id]
-        self.groups[group1_id] = Group(
-            group1_id, group1.stones | group2.stones, group1.liberties, group1.color)
-        del self.groups[group2_id]
-        for s in group2.stones:
-            self.group_index[s] = group1_id
-
-        self._update_liberties(
-            group1_id, add=group2.liberties, remove=group2.stones)
-
-        return group1
+        result = Group(
+            self.max_group_id,
+            frozenset(stones | {played}),
```

Comment:
this would just become `frozenset(stones)` with the above changes.

Reply:
done